src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/DynamicConfigurationCacheProxyTest.java

```diff
@@ -0,0 +1,106 @@
+     * @throws Exception if so.
+     */
+    @Before
+    public void setUp() throws Exception {
+        DynamicConfigurationCacheProxy.getInstance().clear();
+        PowerMockito.mockStatic(GerritDynamicUrlProcessor.class);
+    }
+
+    /**
+     * Sets refresh interval.
+     * @param refreshInternal refresh interval for dynamic trigger
+     */
+    private void setRefreshInternal(long refreshInternal) {
+        PowerMockito.mockStatic(GerritTriggerTimer.class);
+        GerritTriggerTimer timer = mock(GerritTriggerTimer.class);
+        when(GerritTriggerTimer.getInstance()).thenReturn(timer);
+        when(timer.calculateAverageDynamicConfigRefreshInterval()).thenReturn(refreshInternal);
+    }
+
+    /**
+     * Tests the case when cache is empty.
+     * @throws Exception if so.
+     */
+    @Test
+    public void fetchDirectlyWithoutCache() throws Exception {
+        List<GerritProject> expected = Collections.singletonList(mock(GerritProject.class));
+        when(GerritDynamicUrlProcessor.fetch(anyString())).thenReturn(expected);
+
+        List<GerritProject> actual = DynamicConfigurationCacheProxy.getInstance().fetchThroughCache("someUrl");
+
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * Tests the case when cache is non-empty, but record is outdated.
+     * @throws Exception if so.
+     */
+    @Test
+    public void fetchDirectlyWithCache() throws Exception {
+        List<GerritProject> gerritProjects1 = Collections.singletonList(mock(GerritProject.class));
+        List<GerritProject> gerritProjects2 = Collections.singletonList(mock(GerritProject.class));
+        when(GerritDynamicUrlProcessor.fetch(anyString())).thenReturn(gerritProjects1).thenReturn(gerritProjects2);
+        setRefreshInternal(FORCE_REFRESH_INTERVAL);
+
+        List<GerritProject> res1 = DynamicConfigurationCacheProxy.getInstance().fetchThroughCache("someUrl");
+        List<GerritProject> res2 = DynamicConfigurationCacheProxy.getInstance().fetchThroughCache("someUrl");
+
+        assertNotEquals(res1, res2);
+        assertEquals(gerritProjects1, res1);
+        assertEquals(gerritProjects2, res2);
+        verifyStatic(times(2));
+    }
+
+    /**
+     * Tests the case then cache is non-empty and record is still valid.
+     * @throws Exception if so.
+     */
+    @Test
+    public void fetchThroughCache() throws Exception {
+        PowerMockito.mockStatic(GerritDynamicUrlProcessor.class);
+        List<GerritProject> gerritProjects1 = Collections.singletonList(mock(GerritProject.class));
+        List<GerritProject> gerritProjects2 = Collections.singletonList(mock(GerritProject.class));
+        when(GerritDynamicUrlProcessor.fetch(anyString())).thenReturn(gerritProjects1).thenReturn(gerritProjects2);
+
+        List<GerritProject> res1 = DynamicConfigurationCacheProxy.getInstance().fetchThroughCache("someUrl");
+        List<GerritProject> res2 = DynamicConfigurationCacheProxy.getInstance().fetchThroughCache("someUrl");
+
```

Comment:
Should verify that `GerritDynamicUrlProcessor.fetch(` was called on the first call but not the second If I've understood the test correctly.

Reply:
Yes, this is done via `verifyStatic()` that is shortcut for `verifyStatic(times(1))`. I also found the name of this check a bit ... misleading.